### ISSUE TITLE
[Bug] Variables in loop are not detected as used

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#868]
If the variable is at the end of the loop, it will not be detected as used:

```
${counter}    Set Variable    1
WHILE    ${counter} < 10
    Log To Console    ${counter}
    ${counter}    Evaluate ${counter} + 1
END
```

Same for FOR loops. Robocop should imedietely mark such variable as used if it's in the loop and was already used in current scope.
Be careful to still catch following issue:

```
WHILE    ${counter} < 10
    ${not_used}    Keyword
END
```

**Comment by gs-grmadhu** [comment]
Hi @bhirsz ,

When user send ${counter} as an argument then we are observing this issue again. Below is the code snipped along with package versions. Request your help to resolve this issue. Thank you

`Unused Variable Rule Validation
    [Arguments]    ${counter}
    WHILE    ${counter} < 10
        Log To Console    ${counter}
        ${counter}    Evaluate    ${counter} + 1
    END`


robotframework-robocop               4.1.1
robotframework                       5.0.1
Python 3.9.10

**Comment by bhirsz** [comment]
@gs-grmadhu 

Thanks for the report, I will check it.

**Comment by bhirsz** [comment]
I can reproduce the bug - I fixed it, it will be part of the next release.

**Comment by bhirsz** [comment]
Released in 4.2.0

**Comment by gs-skkavuri** [comment]
Hi @bhirsz ,

 we are observing this issue again. Below is the code snippet along with package versions. Request your help to resolve this issue. Thank you

```
*** Settings ***
Documentation    Testing the unused variable
Library    Collections

*** Variables ***
${count}    4

*** Keywords ***
Testing Unused Variable
    [Arguments]    ${list} 
    FOR    ${itemObject}    IN    @{list}
        Set To Dictionary    ${itemObject}    displayOrder=${count}
        ${count}    Evaluate    ${count} + 1
    END
```
robotframework-robocop 4.2.0
robotframework 5.0.1
Python 3.9.10

**Comment by bhirsz** [comment]
Thanks for testing it quickly - most likely I didn't cover FOR loop arguments in the change. I will fix it.

**Comment by bhirsz** [comment]
@gs-skkavuri 

Fixed the issue in the Robocop 4.2.1. Let me know if there are still some issues with this rule - the implementation is due to refactor, but we should at least work on bugs. With every issue our regression test base grows so we at least shouldn't have the same issue again, just similar or uncatched edge cases.
